pkg/fleets/controller.go

```diff
@@ -309,16 +314,102 @@ func (c *Controller) deleteEmptyGameServerSets(fleet *stablev1alpha1.Fleet, list
 }
 
 // recreateDeployment applies the recreate deployment strategy to all non-active
-// GameServerSets
-func (c *Controller) recreateDeployment(list []*stablev1alpha1.GameServerSet) error {
-	for _, gsSet := range list {
+// GameServerSets, and return the replica count for the active GameServerSet
+func (c *Controller) recreateDeployment(fleet *stablev1alpha1.Fleet, rest []*stablev1alpha1.GameServerSet) (int32, error) {
+	for _, gsSet := range rest {
 		if gsSet.Spec.Replicas != 0 {
 			c.logger.WithField("gameserverset", gsSet.ObjectMeta.Name).Info("applying recreate deployment: scaling to 0")
 			gsSetCopy := gsSet.DeepCopy()
 			gsSetCopy.Spec.Replicas = 0
+			if _, err := c.gameServerSetGetter.GameServerSets(gsSetCopy.ObjectMeta.Namespace).Update(gsSetCopy); err != nil {
+				return 0, errors.Wrapf(err, "error updating gameserverset %s", gsSetCopy.ObjectMeta.Name)
+			}
+			c.recorder.Eventf(fleet, corev1.EventTypeNormal, "ScalingGameServerSet",
+				"Scaling inactive GameServerSet %s from %d to %d", gsSetCopy.ObjectMeta.Name, gsSet.Spec.Replicas, gsSetCopy.Spec.Replicas)
+		}
+	}
+
+	return fleet.LowerBoundReplicas(fleet.Spec.Replicas - stablev1alpha1.SumStatusAllocatedReplicas(rest)), nil
+}
+
+// rollingUpdateDeployment will do the rolling update of the old GameServers
+// through to the new ones, based on the fleet.Spec.Strategy.RollingUpdate configuration
+// and return the replica count for the active GameServerSet
+func (c *Controller) rollingUpdateDeployment(fleet *stablev1alpha1.Fleet, active *stablev1alpha1.GameServerSet, rest []*stablev1alpha1.GameServerSet) (int32, error) {
```

Comment:
@enocom particularly love your eyes on this function, and the functions it calls - it's a bit complicated, so if you can see anything that could improve the code / readability, would definitely be interested.